### PR TITLE
Re-arrange the folders, and add peppi to the Galileo notebook

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary
Add peppi to the Galileo notebook.
Organize all the notebook into their respective discipline node.

### CODE REVIEW

**Review by nutjob4life — APPROVED** [review]
Approved based on visual inspection only—trust accordingly

(`brew install gdal` also install `llvm` which is a 2GB (👀) dependency!)

**Review by nutjob4life — CHANGES_REQUESTED** [review]
There seems to be an error in `requirements.txt`:
```console
$ pip install --requirement requirements.txt
ERROR: Invalid requirement: 'requires gdal > 3.8, install on macos with brew install gdal': Expected end or semicolon (after name and no valid version specifier)
    requires gdal > 3.8, install on macos with brew install gdal
             ^ (from line 20 of requirements.txt)
```

**Comment by tloubrieu-jpl** [comment]
Thanks for catching the error on the requirements.txt file @nutjob4life . I fixed that and some other errors that I noticed yesterday while demoing the notebook.

**Review by nutjob4life — COMMENTED** [review]
@tloubrieu-jpl thanks!

I think you might want to add `.spice_kernels/` to `.gitignore`.

I can't get this notebook `GLL_MagPlot.ipynb` to run, though:
```
File ~/Documents/Clients/JPL/PDS/Development/nasa-pds/search-api-notebook/.venv/lib/python3.9/site-packages/pds4_tools/viewer/core.py:21
     18 from ..utils.logging import logger_init
     20 from ..extern import six
---> 21 from ..extern.six.moves.tkinter import Event as TKEvent
     22 from ..extern.six.moves.tkinter import (Tk, Toplevel, PhotoImage, Menu, Scrollbar, Canvas, Frame, Label,
     23                                         Entry, Text, Button, Checkbutton, BooleanVar, StringVar, TclError)
     25 # NOTE: DO NOT IMPORT MATPLOTLIB HERE, or import any other module
     26 # that in-turn imports MPL or any module that imports anything that
     27 # should not be imported when the Viewer is not actually used. See
   (...)
     31 
     32 # Initialize the logger

File ~/Documents/Clients/JPL/PDS/Development/nasa-pds/search-api-notebook/.venv/lib/python3.9/site-packages/pds4_tools/extern/six.py:203, in _SixMetaPathImporter.load_module(self, fullname)
    201 mod = self.__get_module(fullname)
    202 if isinstance(mod, MovedModule):
--> 203     mod = mod._resolve()
    204 else:
    205     mod.__loader__ = self

File ~/Documents/Clients/JPL/PDS/Development/nasa-pds/search-api-notebook/.venv/lib/python3.9/site-packages/pds4_tools/extern/six.py:115, in MovedModule._resolve(self)
    114 def _resolve(self):
--> 115     return _import_module(self.mod)

File ~/Documents/Clients/JPL/PDS/Development/nasa-pds/search-api-notebook/.venv/lib/python3.9/site-packages/pds4_tools/extern/six.py:82, in _import_module(name)
     80 def _import_module(name):
     81     """Import module, returning the module after the last dot."""
---> 82     __import__(name)
     83     return sys.modules[name]

File /opt/homebrew/Cellar/python@3.9/3.9.20/Frameworks/Python.framework/Versions/3.9/lib/python3.9/tkinter/__init__.py:37
     34 import sys
     35 import types
---> 37 import _tkinter # If this fails your Python may not be configured for Tk
     38 TclError = _tkinter.TclError
     39 from tkinter.constants import *

ModuleNotFoundError: No module named '_tkinter'
```
I must doing something wrong, but I don't have time to check into it right now.

**Review by jordanpadams — CHANGES_REQUESTED** [review]
GLL notebook doesn't work. See earlier comments

**Comment by tloubrieu-jpl** [comment]
Ok thanks for the feedback, I ll investigate that from a fresh virtual environment.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Jordan Padams ***@***.***>
Sent: Monday, December 9, 2024 6:00:45 PM
To: NASA-PDS/search-api-notebook ***@***.***>
Cc: Loubrieu, Thomas G (US 398F) ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [NASA-PDS/search-api-notebook] Re-arrange the folders, and add peppi to the Galileo notebook (PR #80)


@jordanpadams requested changes on this pull request.

GLL notebook doesn't work. See earlier comments

File: notebooks/img/leafmap_mess.ipynb [hunk 1]
```diff
@@ -2,7 +2,7 @@
  "cells": [
```
Comment (collinss-jpl) [inline notebooks/img/leafmap_mess.ipynb:2]: <pre class="ql-syntax" spellcheck="false">Line #36.    #polygons = df_4001.apply(polygon, axis=1)
</pre><p>Should just delete these if they're not used anymore. If its something a user of the notebook would find useful by un-commenting, then that should be documented here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NASA-PDS/search-api-notebook/pull/80/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Comment by tloubrieu-jpl** [comment]
@nutjob4life @jordanpadams , regarding the `tinker` import error, did you follow this step from the README ?
image

I am wondering if that would solve your issue.

**Comment by nutjob4life** [comment]
@tloubrieu-jpl thanks for pointing that out. I totally missed those instructions (I saw "at this point, you'll have a locally running…" and assumed I was ready to try it out).

However, even after running the two `labextension install` commands and the `lab build` command—and opening up the AAS WorldWide Telescope tab—I still get the `_tkinter` error when running the notebook.

I vaguely recall from experience that `_tkinter` is Python's Tcl/Tk integration for UI elements. I'm going to try to `brew install python-tk` to see if that helps. Do you have python-tk installed?

**Comment by nutjob4life** [comment]
@tloubrieu-jpl  okay, doing `brew install python-tk@VERSION` got it past the `_tkinter` error. 

The next error is `NameError: name 'eph' is not defined`, but that's under the "work in progress" heading so I'm not concerned about that. Maybe we should comment-out that whole section just to prevent end-users from seeing red?

**Comment by jordanpadams** [comment]
@nutjob4life commented out. merging.

File: notebooks/img/leafmap_mess.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jordanpadams) [reply notebooks/img/leafmap_mess.ipynb:2]: resolved per https://github.com/NASA-PDS/search-api-notebook/pull/80/commits/01e5715ad91b5448c13758670f606620c8bf81bd

Comment (jordanpadams) [reply notebooks/img/leafmap_mess.ipynb:2]: resolved per https://github.com/NASA-PDS/search-api-notebook/pull/80/commits/01e5715ad91b5448c13758670f606620c8bf81bd

**Comment by nutjob4life** [comment]
@jordanpadams thanks much!

**Pull request merged**